File: examples/javascript/client/proto/AckScenePlayers.js
module.exports = AckScenePlayers;

var Packet = require('../net/Packet');
var MsgScenePlayer = require('./MsgScenePlayer');


function AckScenePlayers() {
	this._players = new Array();


	this.Decode = function(packet) {
		var players_count = packet.ReadUshort();
		for (var i = 0; i < players_count; i++)
		{
			this._players.push(new MsgScenePlayer(packet));
		}
	}


	this.SetPlayers = function(players) {
		this._players = players;
	}
	this.GetPlayers= function() {
		return this._players;
	}
}

File: examples/golang/client/proto/ack_scene_players.go
package proto

import (
	"packet"
)

type AckScenePlayers struct {
	Players                  []*MsgScenePlayer
}

func AckScenePlayersDecode(pack *packet.Packet) *AckScenePlayers {
	ackScenePlayers := &AckScenePlayers{}

	PlayersCount := pack.ReadUint16()
	for ;PlayersCount > 0; PlayersCount-- {
		ackScenePlayers.Players = append(ackScenePlayers.Players, MsgScenePlayerDecode(pack))
	}
	return ackScenePlayers
}
